scipy/stats/_qmc.py

```diff
@@ -290,7 +328,10 @@ def discrepancy(sample, *, iterative=False, method="CD", workers=1):
                          f" {set(methods)!r}")
 
 
-def update_discrepancy(x_new, sample, initial_disc):
+def update_discrepancy(
+        x_new: npt.ArrayLike,
+        sample: npt.ArrayLike,
+        initial_disc: float) -> float:
```

Comment:
This function is public so needs this as well
```suggestion
        initial_disc: DecimalNumber) -> float:
```